src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -388,12 +418,32 @@ def _create_request_body(
     """
     request_size_limit = max_error_size if any(map(is_span_has_error, msgs)) else max_size
 
+    size: int = get_event_base64_size(msgs)
     if not prune_size_flag or (
-        len(msgs) < NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION
-        and get_event_base64_size(msgs) < request_size_limit  # noqa
+        len(msgs) < NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION and size < request_size_limit  # noqa
     ):
         return aws_dump(msgs)[:request_size_limit]
 
+    # Process spans: if should_try_zip is True, split and zip the spans, check their size,
+    # and either return the zipped bulks or continue processing.
+    # Also we do not do zip for China region.
+    if should_try_zip and region != CHINA_REGION:
+        get_logger().debug(
+            f"Spans are too big, size [{size}], [{len(msgs)}] spans, bigger than: [{request_size_limit}], trying to split and zip"
+        )
+        zipped_spans_bulks = _split_and_zip_spans(msgs)
+        are_all_spans_small_enough = all(
+            len(zipped_span) <= request_size_limit for zipped_span in zipped_spans_bulks
+        )
+
+        if are_all_spans_small_enough:
+            get_logger().debug(f"Created {len(zipped_spans_bulks)} bulks of zipped spans")
+            return zipped_spans_bulks
+        else:
+            # Continue trimming spans logic goes here
+            get_logger().debug("Some spans are still too large, further trimming may be needed.")
+            pass
```

Comment:
Let's add a try catch block on this, in case zipping fails we want to log this and fallback to the normal span sending

Reply:
Done